applications/StructuralMechanicsApplication/custom_constitutive/linear_elastic_orthotropic_2D_law.cpp

```diff
@@ -0,0 +1,422 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Peter Wilson
+//       Contact:    A.Winterstein@tum.de
+//
+
+// System includes
+#include <iostream>
+
+// External includes
+#include<cmath>
+
+// Project includes
+#include "includes/properties.h"
+#include "custom_constitutive/linear_elastic_orthotropic_2D_law.hpp"
+
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+	//******************************CONSTRUCTOR*********************************
+	//**************************************************************************
+
+	LinearElasticOrthotropic2DLaw::LinearElasticOrthotropic2DLaw()
+		: ConstitutiveLaw()
+	{
+	}
+
+	//******************************COPY CONSTRUCTOR****************************
+	//**************************************************************************
+
+	LinearElasticOrthotropic2DLaw::LinearElasticOrthotropic2DLaw
+		(const LinearElasticOrthotropic2DLaw& rOther)
+		: ConstitutiveLaw(rOther)
+		, mInverseDeformationGradientF0(rOther.mInverseDeformationGradientF0)
```

Comment:
why do you store this? if it is linear elastic you don't need to store any of those.

in particular the strain energy should not be stored, instead the appropriate calculate functions should be called when needed